lib/segment/src/utils/mem.rs

```diff
@@ -0,0 +1,189 @@
+use crate::entry::entry_point::{OperationError, OperationResult};
+
+#[derive(Debug)]
+pub struct Mem {
+    #[cfg(target_os = "linux")]
+    cgroups: Option<cgroups_mem::CgroupsMem>,
+    sysinfo: sysinfo_mem::SysinfoMem,
+}
+
+impl Mem {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            #[cfg(target_os = "linux")]
+            cgroups: cgroups_mem::CgroupsMem::new(),
+            sysinfo: sysinfo_mem::SysinfoMem::new(),
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn refresh(&mut self) {
+        #[cfg(target_os = "linux")]
+        if let Some(cgroups) = &mut self.cgroups {
+            cgroups.refresh();
+        }
+
+        self.sysinfo.refresh();
+    }
+
+    pub fn total_memory_bytes(&self) -> u64 {
+        #[cfg(target_os = "linux")]
+        if let Some(cgroups) = &self.cgroups {
+            if let Some(memory_limit_bytes) = cgroups.memory_limit_bytes() {
+                return memory_limit_bytes;
+            }
+        }
+
+        self.sysinfo.total_memory_bytes()
+    }
+
+    pub fn available_memory_bytes(&self) -> u64 {
+        #[cfg(target_os = "linux")]
+        if let Some(cgroups) = &self.cgroups {
+            if let Some(memory_limit_bytes) = cgroups.memory_limit_bytes() {
+                return memory_limit_bytes.saturating_sub(cgroups.used_memory_bytes());
```

Comment:
Are in memory buffers and cache indeed excluded from the `used_memory_bytes()` value? I believe it is, just want to confirm.